src/Fields/FieldArrays.jl

```diff
@@ -664,3 +664,37 @@ for op in (:*,:⋅,:⊙,:⊗)
     end
   end
 end
+
+# Inverse fields
+
+struct InverseField{F} <: Field
+  original::F
+end
+
+inverse_map(a::Field) = InverseField(a)
+inverse_map(a::InverseField) = a.original
+
+function return_cache(a::InverseField,x::Point)
+  return return_cache(a.original,x), return_cache(∇(a.original),x)
+end
+
+function evaluate!(caches,a::InverseField,x::Point)
+  P = typeof(x)
+  D = length(x)
+  cache,∇cache = caches
+  # Initial guess. (Can this be improved?)
+  y₀ = zero([x...])
```

Comment:
Use this to avoid allocations: y0 = SVector(Tuple((zero(P))))

Reply:
Unfortunately, NLsolve requires a mutable vector, and MVector does not work. We're stuck with a regular array.

I couldn't find any good alternatives to NLsolve either.

Reply:
This should work (at least in the Gridap part), if not i am overlooking something.
```julia
  y₀ = MVector(Tuple(zero(P)))
  f!(F,y) = F .= SVector{D}(Tuple(evaluate!(cache,a.original,P(Tuple(y))) - x))
  j!(J,y) = J .= SMatrix{D,D}(Tuple(evaluate!(∇cache,∇(a.original),P(Tuple(y)))))
```

Reply:
Using an `MVector` leads to a nan in NLsolve at run time, for a very simple test case with an affine function. This nan then makes the solver abort without converging.

Using a Julia array does not encounter a nan. I assume that NLsolver doesn't handle `MVector`s correctly for some reason.

Reply:
OK! Then allocate the vector `y0` in the cache.

Reply:
Done. I also allocate `F0` in the cache.